src/nvim/mbyte.c

```diff
@@ -118,6 +119,32 @@ struct interval {
 #endif
 
 /*
+ * To speed up BYTELEN() we fill a table with the byte lengths whenever
+ * enc_utf8 or enc_dbcs changes.
+ */
+static char mb_bytelen_tab[256];
```

Comment:
`mb_bytelen_tab` does not contain characters. Use `int8_t` or, better, `uint8_t` (because length cannot be negative).